### ISSUE TITLE
Both hlpn and pthlpng plugins register a package for 'http:///hlpn.integers.ecore'

### ISSUE THREAD

**Issue by abelgomez** [lip6/pnmlframework#10]
Both `fr.lip6.pnml.framework.hlpn` and `fr.lip6.pnml.framework.pthlpng` plugins register a package for `http:///hlpn.integers.ecore`.

This initially causes a warning from EMF, and later, if the EMF package registry is processed by other tools (such as the QVTo engine), causes a ClassCastException (java.lang.ClassCastException: fr.lip6.move.pnml.pthlpng.integers.impl.IntegersFactoryImpl cannot be cast to fr.lip6.move.pnml.hlpn.integers.IntegersFactory).

```
!SESSION 2016-06-27 17:12:55.285 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.8.0_77
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=es_ES
Framework arguments:  -product org.eclipse.sdk.ide
Command-line arguments:  -product org.eclipse.sdk.ide -data C:\Eclipse\eclipse-SDK-4.6-win32-x86_64-Papyrus\eclipse\workspace/../runtime-EclipseApplication -dev file:C:/Eclipse/eclipse-SDK-4.6-win32-x86_64-Papyrus/eclipse/workspace/.metadata/.plugins/org.eclipse.pde.core/Eclipse Application/dev.properties -os win32 -ws win32 -arch x86_64 -consoleLog

!ENTRY org.eclipse.emf.ecore 2 0 2016-06-27 17:13:00.667
!MESSAGE Both 'fr.lip6.pnml.framework.hlpn' and 'fr.lip6.pnml.framework.pthlpng' register a package for 'http:///hlpn.integers.ecore'

!ENTRY org.eclipse.egit.ui 2 0 2016-06-27 17:13:10.474
!MESSAGE Warning: The environment variable HOME is not set. The following directory will be used to store the Git
user global configuration and to define the default location to store repositories: 'C:\Users\agomez'. If this is
not correct please set the HOME environment variable and restart Eclipse. Otherwise Git for Windows and
EGit might behave differently since they see different configuration options.
This warning can be switched off on the Team > Git > Confirmations and Warnings preference page.

!ENTRY org.eclipse.emf.ecore 2 0 2016-06-27 17:13:15.326
!MESSAGE fr.lip6.move.pnml.pthlpng.integers.impl.IntegersFactoryImpl cannot be cast to fr.lip6.move.pnml.hlpn.integers.IntegersFactory
!STACK 0
java.lang.ClassCastException: fr.lip6.move.pnml.pthlpng.integers.impl.IntegersFactoryImpl cannot be cast to fr.lip6.move.pnml.hlpn.integers.IntegersFactory
    at fr.lip6.move.pnml.hlpn.integers.impl.IntegersFactoryImpl.init(IntegersFactoryImpl.java:71)
    at fr.lip6.move.pnml.hlpn.integers.IntegersFactory.<clinit>(IntegersFactory.java:51)
    at fr.lip6.move.pnml.hlpn.integers.impl.IntegersPackageImpl.<init>(IntegersPackageImpl.java:212)
    at fr.lip6.move.pnml.hlpn.integers.impl.IntegersPackageImpl.init(IntegersPackageImpl.java:240)
    at fr.lip6.move.pnml.hlpn.integers.IntegersPackage.<clinit>(IntegersPackage.java:87)
    at fr.lip6.move.pnml.hlpn.finiteIntRanges.impl.FiniteIntRangesPackageImpl.init(FiniteIntRangesPackageImpl.java:201)
    at fr.lip6.move.pnml.hlpn.finiteIntRanges.FiniteIntRangesPackage.<clinit>(FiniteIntRangesPackage.java:87)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.finiteEnumerations.impl.FiniteEnumerationsPackageImpl.init(FiniteEnumerationsPackageImpl.java:152)
    at fr.lip6.move.pnml.hlpn.finiteEnumerations.FiniteEnumerationsPackage.<clinit>(FiniteEnumerationsPackage.java:87)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.dots.impl.DotsPackageImpl.init(DotsPackageImpl.java:147)
    at fr.lip6.move.pnml.hlpn.dots.DotsPackage.<clinit>(DotsPackage.java:85)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.cyclicEnumerations.impl.CyclicEnumerationsPackageImpl.init(CyclicEnumerationsPackageImpl.java:163)
    at fr.lip6.move.pnml.hlpn.cyclicEnumerations.CyclicEnumerationsPackage.<clinit>(CyclicEnumerationsPackage.java:87)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.booleans.impl.BooleansPackageImpl.init(BooleansPackageImpl.java:200)
    at fr.lip6.move.pnml.hlpn.booleans.BooleansPackage.<clinit>(BooleansPackage.java:86)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.arbitrarydeclarations.impl.ArbitrarydeclarationsPackageImpl.init(ArbitrarydeclarationsPackageImpl.java:157)
    at fr.lip6.move.pnml.hlpn.arbitrarydeclarations.ArbitrarydeclarationsPackage.<clinit>(ArbitrarydeclarationsPackage.java:87)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at fr.lip6.move.pnml.hlpn.hlcorestructure.impl.HlcorestructurePackageImpl.init(HlcorestructurePackageImpl.java:511)
    at fr.lip6.move.pnml.hlpn.hlcorestructure.HlcorestructurePackage.<clinit>(HlcorestructurePackage.java:87)
    at sun.misc.Unsafe.ensureClassInitialized(Native Method)
    at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(Unknown Source)
    at sun.reflect.ReflectionFactory.newFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.acquireFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.getFieldAccessor(Unknown Source)
    at java.lang.reflect.Field.get(Unknown Source)
    at org.eclipse.emf.ecore.plugin.RegistryReader$EPackageDescriptor.getEPackage(RegistryReader.java:275)
    at org.eclipse.emf.ecore.impl.EPackageRegistryImpl.getEPackage(EPackageRegistryImpl.java:127)
    at org.eclipse.m2m.internal.qvt.oml.blackbox.java.JavaModuleLoader.resolvePackages(JavaModuleLoader.java:156)
    at org.eclipse.m2m.internal.qvt.oml.blackbox.java.JavaModuleLoader.loadModule(JavaModuleLoader.java:88)
    at org.eclipse.m2m.internal.qvt.oml.blackbox.java.JavaBlackboxProvider.loadCompilationUnit(JavaBlackboxProvider.java:89)
    at org.eclipse.m2m.internal.qvt.oml.blackbox.BlackboxRegistry.loadCompilationUnit(BlackboxRegistry.java:68)
    at org.eclipse.m2m.internal.qvt.oml.compiler.BlackboxUnitResolver$BBoxUnitContents.loadElements(BlackboxUnitResolver.java:131)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.loadBlackboxUnit(QVTOCompiler.java:563)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.doCompile(QVTOCompiler.java:354)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.doCompile(QVTOCompiler.java:410)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.compileSingleFile(QVTOCompiler.java:318)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.compile(QVTOCompiler.java:205)
    at org.eclipse.m2m.internal.qvt.oml.compiler.QVTOCompiler.compile(QVTOCompiler.java:218)
    at org.eclipse.m2m.internal.qvt.oml.editor.ui.QvtCompilerFacade.compile(QvtCompilerFacade.java:79)
    at org.eclipse.m2m.internal.qvt.oml.editor.ui.QvtReconcilingStrategy.getCompilationResult(QvtReconcilingStrategy.java:90)
    at org.eclipse.m2m.internal.qvt.oml.editor.ui.QvtReconcilingStrategy.reconcileInternal(QvtReconcilingStrategy.java:74)
    at org.eclipse.m2m.internal.qvt.oml.editor.ui.QvtReconcilingStrategy.initialReconcile(QvtReconcilingStrategy.java:62)
    at org.eclipse.jface.text.reconciler.MonoReconciler.initialProcess(MonoReconciler.java:95)
    at org.eclipse.jface.text.reconciler.AbstractReconciler$BackgroundThread.run(AbstractReconciler.java:174)


```

**Comment by yanntm** [comment]
hi,
pretty old PR  ! sorry.
The problem is that there is both correct semantic changes and stuff that is useless e.g. whitespace differences or generated files that vary slightly in insignificant ways.
Could you perhaps build a simpler PR, where *only* the bad namespace is changed ?
thank you.

**Comment by yanntm** [comment]
It's ok if you only edit the generated classes "in place" like I did on the UTF-8 patch, we are reluctant to regenerate the whole sources at this point; if your patch also edits generated files we'll take that too, but not edits to generated files.

**Comment by abelgomez** [comment]
Since I don't want to break anything (and to be fair, don't have a dev environment to do it), maybe it's just easier to explain the issue.

The only manual change needed is in file 
https://github.com/lip6/pnmlframework/blob/346b9f9005d5dd4c6fc6f5034caf58c583b21948/pnmlFw-Low_Level_API_Generation/model/pthlpng/integers.ecore#L3

The `nsURI="http:///hlpn.integers.ecore"` attribute should be `nsURI="http:///pthlpng.integers.ecore" `.

All the other changes happen in automatically generated files.

**Comment by yanntm** [comment]
Ok... would it be ok to simply replace occurrences of string http:///hlpn.integers.ecore by http:///pthlpng.integers.ecore or does this setting really impact the generated code in other ways ?

**Comment by abelgomez** [comment]
I'd say that manually changing automatically generated code (rather than regenerating it from the source model), it's really a bad idea. Nevertheless, let me do a manual inspection of the files of my PR to refresh my memory, and I'll give you an answer.

**Comment by yanntm** [comment]
meh, ok that is not enough, the whole generated files need to be patched to have Java correspondance correct it seems. I'm still thinking of just editing generated files however, simply track down the whole pthlpng.Int references and refactor to hlpn.
What a pain, is this still useful for you ?
I guess you solved the issue on your fork and survived not getting an answer here, and maybe defended your PhD since :D 
If not, I think I'll just let the issue simmer happily, ppl can have a look at your fork if they meet the same issue (or that day I'll take the time to patch it).

**Comment by abelgomez** [comment]
I'm pretty sure that, besides the manual modification to `integers.ecore`, the only "real" modification to the generated sources affect `plugin.xml` and `IntegersPackage.java`.

I'll try to generate a new PR with less code modifications in the next days if you feel more comfortable with it.

Regarding "my fork", what I finally did is to avoid loading both the plugins for _hlpn_ and _pthlpng_ Petri nets from the official repo to avoid the name collision (so there was no need to keep my separate fork), but still, I think this is something should be fixed :)

Aaand, no this wasn't needed for my PhD, I just like to help others and contribute to OSS ;)

**Comment by yanntm** [comment]
a smaller patch request if you had the time to format it and base yourself off the latest master would be accepted for sure, thank you !
I also have never loaded both plugins simultaneously in my usage of PNMLFW.
But I agree this should be patched, might as well go through with it now you've tagged the issue and identified root cause.

**Comment by yanntm** [comment]
> I'd say that manually changing automatically generated code (rather than regenerating it from the source model), it's really a bad idea.
yes if travis was properly regenerating the code and it wasn't under git storage, but since the current rebuild does not replay the EMFgen part I'm too afraid of things breaking, the last we generated these files was *long ago* by EMF/eclipse progress rate in this period.

**Comment by abelgomez** [comment]
Just to keep all the info w.r.t. this issue: the large amount of changes that apparently the new code generation causes is not caused by a change in EMF nor in the templates, it's just a formatting issue. Eclipse changed twice the code formatter since the first code generation of PNMLfw (see https://github.com/alostale/luna-java-formatter).

A manual regeneration and a careful code inspection should show that the actual number of changes in the code are minimal.

**Comment by yanntm** [comment]
hmm;

maybe if we right click the whole project, go Source->Format, commit these whitespace changes, and then in separately with the result of the change to the grammar uri that would produce a small readable second commit, and a big unreadable/auditable but trusted to be whitespace only first commit.

**Comment by abelgomez** [comment]
That's exactly what I'm doing roght now ;) keep you posted

**Comment by yanntm** [comment]
otherwise if you have manually audited the changes and saw nothing really semantic/game breaking, I think we'll adopt your PR anyway if it's fast-forward on git.

**Comment by abelgomez** [comment]
Since the code changed quite a bit, closed the old PR #11 and opened a new one #21. Now the inspection of code changes is much more manageable. There are still quite files involved, but most of the changes affect comments and it is much easier to follow the modifications.
